### PR TITLE
Add contextual searches to playlists, library and global search.

### LINKED ISSUE #5567: Enable search-by-artist and search-by-album when right-clicking on a song in playlist or search.
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [X] I checked the issue tracker for similar issues
- [X] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [X] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Arch
- Clementine version: 1.3.1

### Expected behaviour / actual behaviour

Clementine should allow the user to search by artist or search by album by right-clicking on a search result, or a song in a playlist, and selecting the corresponding option. 

Example: User searches for song X, knowing it's on album Y. Instead of having to then search for album Y manually, the user right clicks on the result for song X, and selects "search by/for album", which returns the entire album.

### PR DESCRIPTION
**Pull request by JacobHenner**
Closes #5567

### CODE REVIEW

File: src/globalsearch/globalsearchview.cpp [hunk 1]
```diff
@@ -515,6 +525,32 @@ void GlobalSearchView::OpenSelectedInNewPlaylist() {
   emit AddToPlaylist(data);
 }
 
+void GlobalSearchView::SearchForArtist() {
+  if (const SongMimeData* data = qobject_cast<const SongMimeData*>(SelectedMimeData())) {
+    if (data && !data->songs.isEmpty() && !data->songs.first().artist().isEmpty()) {
+      StartSearch(data->songs.first().artist().simplified());
+    }
+  }
```
Comment (hatstand) [inline src/globalsearch/globalsearchview.cpp]: Should be the same line as the `else if`. Run `make format` to fix.

**Comment by JacobHenner** [comment]
I've noticed that this does not work as expected if you select an album rather than a song under the library view. Will investigate sometime this week. This is not an issue with global search.

File: src/globalsearch/globalsearchview.cpp (hunk 1 above)
Comment (JacobHenner) [inline src/globalsearch/globalsearchview.cpp]: `make format` did not fix this for me. I'll do it manually.

**Comment by JacobHenner** [comment]
@hatstand Fixed the issues I was encountering, and the formatting. Should be good to go.

**Comment by JacobHenner** [comment]
Found a minor bug where selecting "search for this" on an album in Library view results in a prefixed entry in the search box. Will investigate.

File: src/library/libraryview.cpp [hunk 2]
```diff
@@ -709,6 +737,29 @@ void LibraryView::FilterReturnPressed() {
...
+  SaveFocus();
+  if (!last_selected_song_.album().isEmpty()) {
+    filter_->ShowInLibrary(last_selected_song_.album().simplified());
+  }
+}
+
+void LibraryView::SearchForThis() {
+  SaveFocus();
+  if (!last_selected_container_.isEmpty()) {
+    filter_->ShowInLibrary(last_selected_container_.simplified());
```
Comment (JacobHenner) [inline src/library/libraryview.cpp]: Is there any way to get a more predictable result for last_selected_container_? Right now, it varies based upon the sorting/grouping method used. For example: 

2chainz

selecting "search for this" results in a search for `2013B.O.A.T.S. II #MeTime`. 

Searching for 
screenshot from 2017-03-14 23-00-40

results in a search for `10302009 i  festival 8 ca`

Comment (JacobHenner) [inline src/library/libraryview.cpp]: Figured it out. Will tackle tomorrow.

Comment (JacobHenner) [inline src/library/libraryview.cpp]: So I see two options here: 
1. Search for exactly what's being displayed when a user clicks "search for this" on a divider, which might not return results if the album year is included as per the sorting preferences.
2. Modify librarymodel.cpp to record the album, album artist, or artist for each divider (depending on type of divider), and search for that. 

Which is more suitable?

**Comment by Fat-Zer** [comment]
If in library view settings "Show only duplicates/untaged" is set thus the Filter field is disabled (why/should it?), but it can still be populated by "Search for *" actions.

**Comment by JacobHenner** [comment]
@Fat-Zer I will take a look at this.

Also, just noticed a bug/inconsistency in global search when right-clicking on a container/divider. Going to bring it in-line with the Library search behavior of "search for this" if possible.

File: src/library/libraryview.cpp (hunk 2 above)
Comment (hatstand) [inline src/library/libraryview.cpp]: It would be pretty weird if a user could initiate a search from there that resulted in zero results.

**Comment by JacobHenner** [comment]
@hatstand Would a schema update be acceptable in order to complete this feature?

**Comment by hatstand** [comment]
If you need it, sure.

**Comment by JacobHenner** [comment]
Updated to show "search for this" instead of album/artist search in both global and library views for a more consistent experience. User can click on headers in tree to search for artist or album, or whatever other attributes user decides to sort tree by.

**Comment by JacobHenner** [comment]
@hatstand Any thoughts?

**Comment by JacobHenner** [comment]
Anyone available to review this?

**Comment by JacobHenner** [comment]
Rebased

**Comment by hatstand** [comment]
I think you forgot to include the new schema

**Comment by JacobHenner** [comment]
@hatstand Indeed. Fixing now, standby.

**Comment by JacobHenner** [comment]
@hatstand Fixed.

**Comment by JacobHenner** [comment]
@hatstand  This should be ready

**Comment by hatstand** [comment]
I still don't see a schema

**Comment by JacobHenner** [comment]
@hatstand Added. Must have gotten lost in a bad rebase, sorry.

**Comment by JacobHenner** [comment]
This should be good to go.

**Comment by hatstand** [comment]
Travis says this doesn't build, please fix.

**Comment by JacobHenner** [comment]
That was embarrassing. Looks like some extraneous text made its way into that commit. It should build now...

**Comment by JacobHenner** [comment]
Yep, it's good.

**Comment by JacobHenner** [comment]
Thank you, sir!

**Pull request merged**